### PR TITLE
centralize notifications in directory service

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public static final String HEADER_ERROR = "error";
+    public static final String HEADER_NOTIFICATION_TYPE = "notificationType";
+    public static final String HEADER_ELEMENT_NAME = "elementName";
+    public static final String HEADER_STUDY_UUID = "studyUuid";
+    private static final String CATEGORY_BROKER_OUTPUT = DirectoryService.class.getName() + ".output-broker-messages";
+    private static final Logger MESSAGE_OUTPUT_LOGGER = LoggerFactory.getLogger(CATEGORY_BROKER_OUTPUT);
+
+    @Autowired
+    private StreamBridge studyUpdatePublisher;
+
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/NotificationService.java:41]: Add  message `PostCompletion` ?

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 2]
```diff
@@ -99,34 +88,16 @@ public Consumer<Message<String>> consumeStudyUpdate() {
...
-            .setHeader(HEADER_USER_ID, userId)
-            .setHeader(HEADER_DIRECTORY_UUID, directoryUuid)
-            .setHeader(HEADER_ELEMENT_NAME, elementName)
-            .setHeader(HEADER_IS_ROOT_DIRECTORY, isRoot)
-            .setHeader(HEADER_IS_PUBLIC_DIRECTORY, isPrivate == null || !isPrivate) // null may only come from borked REST request
-            .setHeader(HEADER_NOTIFICATION_TYPE, notificationType)
-            .setHeader(HEADER_UPDATE_TYPE, UPDATE_TYPE_DIRECTORIES)
-            .setHeader(HEADER_ERROR, error);
-        sendUpdateMessage(messageBuilder.build());
+        notificationService.emitDirectoryChanged(directoryUuid, elementName, userId, null, isPrivate, isRoot, notificationType);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:100]: Remove this method

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/directory/server/NotificationService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/directory/server/NotificationService.java:43]: ok. done.

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/directory/server/DirectoryService.java]: ok. done

**Review by bendaamerahm — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**